tests/std/tests/P0896R4_stream_iterators/test.cpp

```diff
@@ -101,6 +107,78 @@ void test_ostreambuf_iterator(basic_ostream<CharT, Traits>& os) {
 #endif // __cpp_lib_concepts
 }
 
+template <class BidIt, class CharT = typename iterator_traits<BidIt>::value_type, class Traits = regex_traits<CharT>>
+void test_regex_iterator() {
+    using I = regex_iterator<BidIt, CharT, Traits>;
+    STATIC_ASSERT(is_same_v<typename I::difference_type, ptrdiff_t>);
+    STATIC_ASSERT(is_same_v<typename I::iterator_category, forward_iterator_tag>);
+
+#ifdef __cpp_lib_concepts
+    const I i{};
+
+    assert(i == default_sentinel);
+    assert(default_sentinel == i);
+    assert(!(i != default_sentinel));
+    assert(!(default_sentinel != i));
+
+    STATIC_ASSERT(forward_iterator<I>);

```

Comment:
Why?

Reply:
Perhaps because I didn't get the intent of running in older modes at the time I wrote this. I think I should remove these unrelated stuffs now.